Ошибки при развертывании проекта

Последовательно иду по шагам в README.md и получаю ошибки. До шага с миграциями все проходит успешно.

На шаге `./migration/run.sh` ошибка:

    File liquibase.properties not found

`stack run` и `stack exec haskell-starter-kit-exe` оба выдают следующее:

```
{"appInstanceName":"AppName","severity":"Info","message":"Starting application...","timestamp":"2021-03-29T09:45:28.388407263Z","trace":"[Lib.runLogExample#58] "}
{"appInstanceName":"AppName","severity":"Debug","message":"Here is how we work!","timestamp":"2021-03-29T09:45:28.388619931Z","trace":"[Lib.runLogExample#59] "}
{"appInstanceName":"AppName","severity":"Debug","message":"starting server","timestamp":"2021-03-29T09:45:28.388675734Z","trace":"[Lib.logDebug#30] "}
haskell-starter-kit-exe: ParseError "" "no such variable \"DB_HOST\""
```

При этом `stack build` успешно билдит, `make deps` и `make run` тоже работают. docker.io и docker-compose установлены. Файлы `.env` и `config/local.conf` скопированы из темплейтов. Содержимое `.env`:

```
POSTGRES_HOST_AUTH_METHOD=md5
DB_HOST=localhost
DB_PORT=5432
POSTGRES_USER=kav
POSTGRES_PASSWORD=
POSTGRES_DB=appnamedb
```

Содержимое `config/local.conf`:

```
database {
  host = "$(DB_HOST)"
  port = "$(DB_PORT)"
  name = "$(POSTGRES_DB)"
  user = "$(POSTGRES_USER)"
  pass = "$(POSTGRES_PASSWORD)"
  pool_limit = 10
}

web_server {
  port = 8080
}

auth {
  key_path = "./keys/auth-key"
}

log {
  app_instance_name = "AppName"
  log_to_stdout = on
  log_level = "Debug"
}

app {
  domain_specific = "SOME_DOMAIN_SPECIFIC_PARAM"
}
```

Система Linux debian bullseye.

Хорошо, одну проблему решили. В README добавлю строчку про `export $(xargs < .env)`.

Что касается миграций, то в ридми не сказано про копирование файла liquibase.properties, нужно добавить. Однако после копирования и правки liquibase.properties `./migrations/run.sh` все равно не работает, выдает ошибку:

```
Mar 30, 2021 12:29:26 PM org.postgresql.Driver connect
SEVERE: Connection error: 
org.postgresql.util.PSQLException: The connection attempt failed.
        at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:275)
        at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:49)
        at org.postgresql.jdbc.PgConnection.<init>(PgConnection.java:194)
        at org.postgresql.Driver.makeConnection(Driver.java:450)
        at org.postgresql.Driver.connect(Driver.java:252)
        at liquibase.database.DatabaseFactory.openConnection(DatabaseFactory.java:239)
        at liquibase.database.DatabaseFactory.openDatabase(DatabaseFactory.java:151)
        at liquibase.integration.commandline.CommandLineUtils.createDatabaseObject(CommandLineUtils.java:85)
        at liquibase.integration.commandline.Main.doMigration(Main.java:958)
        at liquibase.integration.commandline.Main.run(Main.java:188)
        at liquibase.integration.commandline.Main.main(Main.java:103)
Caused by: java.net.UnknownHostException: host.docker.internal
        at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:184)
        at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
        at java.net.Socket.connect(Socket.java:589)
        at org.postgresql.core.PGStream.<init>(PGStream.java:68)
        at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:144)
        ... 10 more
```

Файл migrations/liquibase.properties:

```
classpath: /opt/jdbc/postgres-jdbc.jar
driver: org.postgresql.Driver
url: jdbc:postgresql://host.docker.internal:5432/appnamedb
username: kav
password:
changeLogFile: changelog.xml
logLevel: info
```

Судя по ошибке, не может найти host.docker.internal. `make run` я предварительно запускал.

--- a/Makefile
+++ b/Makefile
@@ -18,7 +18,7 @@ stop:
 	docker-compose down
 
 prune: stop
-	docker volume rm haskellstarterkit_pg-data
+	docker volume rm haskell-starter-kit_pg-data
 
 style:
 	@cd "$$(git rev-parse --show-toplevel)" && ./scripts/format-code

--- a/README.md
+++ b/README.md
@@ -14,6 +14,11 @@ To configure env for project:
 
 `cp ./.env.template ./.env`
 
+3. Configure liquibase at `migrations/liquibase.properties` using template at
+   `migrations/liquibase.template.properties`:
+
+`cp migrations/liquibase.template.properties migrations/liquibase.properties`
+
 ## Run with docker
 
 You need to have docker installed in your system. Then run:
@@ -22,6 +27,9 @@ You need to have docker installed in your system. Then run:
 make deps
 make run
 ```
+
+`make run` is needed every time you'd like to build and run the project in a
+docker container. `make deps` is only rarely needed when `make run` fails.
 
 ## Develop with stack
 
@@ -37,7 +45,7 @@ To build project:
 
 `stack build`
 
-To run migration:
+To run migration, ensure you have run `make run` and run the command:
 
 `./migration/run.sh`
 

--- a/migrations/liquibase.template.properties
+++ b/migrations/liquibase.template.properties
@@ -1,6 +1,6 @@
 classpath: /opt/jdbc/postgres-jdbc.jar
 driver: org.postgresql.Driver
-url: jdbc:postgresql://host.docker.internal:5432/appnamedb
+url: jdbc:postgresql://localhost:5431/appnamedb
 username: myuser
 password: mypass
 changeLogFile: changelog.xml

--- a/migrations/run.sh
+++ b/migrations/run.sh
@@ -1,6 +1,7 @@
 #!/bin/sh
+
 docker run \
     --network="host" \
-    -v `pwd`/migrations:/workspace/ \
+    -v "$(pwd)"/migrations:/workspace/ \
     kilna/liquibase-postgres \
     liquibase update

